tests/parametric/test_telemetry.py

```diff
@@ -0,0 +1,180 @@
+"""
+Test the telemetry that should be emitted from the library.
+"""
+import time
+import uuid
+
+import pytest
+
+from utils import scenarios, rfc
+
+
+DEFAULT_ENVVARS = {
+    # Decrease the heartbeat/poll intervals to speed up the tests
+    "DD_TELEMETRY_HEARTBEAT_INTERVAL": "0.2",
+}
+
+
+@rfc("https://docs.google.com/document/d/14vsrCbnAKnXmJAkacX9I6jKPGKmxsq0PKUb3dfiZpWE/edit")
+@scenarios.parametric
+class Test_First_Trace_Telemetry:
+    """
+    Test the time-to-first-trace telemetry that should be emitted from the library.
+
+    The time-to-first trace telemetry provides insight into how long it takes for the library to emit its first trace
+    from the time of Agent installation.
+    """
+
+    @pytest.mark.parametrize(
+        "library_env",
+        [
+            {
+                **DEFAULT_ENVVARS,
+                "DD_INSTRUMENTATION_INSTALL_TIME": str(int(time.time())),
+                "DD_INSTRUMENTATION_INSTALL_TYPE": "k8s_single_step",
+                "DD_INSTRUMENTATION_INSTALL_ID": str(uuid.uuid4()),
+            },
+        ],
+    )
+    def test_first_trace_telemetry_propagated(self, library_env, test_agent, test_library):
+        """
+        When instrumentation data is propagated to the library
+            The first trace should contain telemetry for calculating how long it took to emit the first trace.
+        """
+        with test_library.start_span("first_span"):
```

Comment:
What should be the behavior when stats computation is enabled and the first span is not sampled (sent to the agent)?

Reply:
We've intentionally left this as out of scope for now to keep things simple. An improvement could be made to the spec to only tag the first sampled trace. We're not expecting many users to deploy for the first time with sophisticated sampling configurations.

Reply:
👍🏻